DllImportGenerator/DllImportGenerator/ManualTypeMarshallingAnalyzer.cs

```diff
@@ -0,0 +1,342 @@
+using System;
+using System.Collections.Immutable;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+#nullable enable
+
+namespace Microsoft.Interop
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class ManualTypeMarshallingAnalyzer : DiagnosticAnalyzer
+    {
+        private const string DiagnosticIdPrefix = "INTEROPGEN";
+        private const string Category = "Interoperability";
+
+        public readonly static DiagnosticDescriptor BlittableTypeMustBeBlittableRule =
+            new DiagnosticDescriptor(
+                "INTEROPGEN001",
+                "BlittableTypeMustBeBlittable",
+                "Type '{0}' is marked with BlittableTypeAttribute but is not blittable.",
+                Category,
+                DiagnosticSeverity.Error,
+                isEnabledByDefault: true,
+                description: "A type marked with BlittableTypeAttribute must be blittable.");
+
+        public readonly static DiagnosticDescriptor CannotHaveMultipleMarshallingAttributesRule =
+            new DiagnosticDescriptor(
+                "INTEROPGEN002",
+                "CannotHaveMultipleMarshallingAttributes",
+                "Type '{0}' is marked with BlittableTypeAttribute and NativeMarshallingAttribute. A type can only have one of these two attributes.",
```

Comment:
Maybe just add a .resx for everything now, so we don't have to move all the strings everything later.